crates/weaver_resolved_schema/src/body.rs

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(rustdoc::invalid_html_tags)]
+
+//! Specification of a resolved `BodyField`.
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use weaver_semconv::attribute::{AttributeType, Examples, RequirementLevel};
+use weaver_semconv::body::{BodyFieldSpec, BodySpec, BodyType};
+use weaver_semconv::stability::Stability;
+
+/// A `Body` definition.
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq, Hash, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct Body {
+    /// Identifies the type of the body. It can be "map", "string".
+    pub r#type: BodyType,
+    /// A brief description of the body.
+    #[serde(skip_serializing_if = "String::is_empty")]
+    #[serde(default)]
+    pub brief: String,
+    /// A more elaborate description of the body.
+    /// It defaults to an empty string.
+    #[serde(skip_serializing_if = "String::is_empty")]
+    #[serde(default)]
+    pub note: String,
+    /// Specifies the stability of the body.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub stability: Option<Stability>,
+    /// Sequence of example values for the body or single example
+    /// value. They are required only for string types. Example values
+    /// must be of the same type of the body. If only a single example is
+    /// provided, it can directly be reported without encapsulating it
+    /// into a sequence/dictionary.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub examples: Option<Examples>,
+    /// Identifies the definition of the "fields" of the body when the body type is "map".
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub fields: Option<Vec<BodyField>>,
```

Comment:
I think I'd remove the fields from here and encode them in the `BodyType` struct, flattening that into this struct for disjoint mapped union.